src/ui/mainwindow.cpp

```diff
@@ -927,9 +928,7 @@ MainWindow::MainWindow(Application* app, SystemTrayIcon* tray_icon, OSD* osd,
           SLOT(EnableKittens(bool)));
   connect(ui_->action_kittens, SIGNAL(toggled(bool)), app_->network_remote(),
           SLOT(EnableKittens(bool)));
-  QString showConsole =
-      QProcessEnvironment::systemEnvironment().value(kShowDebugConsoleKey, "0");
-  if (showConsole == "1")
+  if (debug_console_.get())
```

Comment:
I think you want to keep the logic here as it was and allow `debug_console_->show()` inside `ShowConsole()` to implicitly create the `Console`.

Reply:
Yeah. My ulterior motive for extending the lifetime of the Console instance is to support an interface where other components (specifically GetEngine) can add their own pages. If this occurred on initialization, the Lazy wrapper would just provide a mechanism to aid in initialization dependency resolution.

Maybe the better option is to emit a signal when a Console is actually created. That would still allow the Lazy construction to remain lazy.

Reply:
Fixed.